state.c

```diff
@@ -515,6 +512,26 @@ plat_goal_state_t pv_state_check_goals(struct pv_state *s,
 	return s_goal_state;
 }
 
+static plat_goal_state_t pv_state_check_goal_prev_group(struct pv_state *s,
+							struct pv_group *group)
+{
+	if (!s || !group)
+		return PLAT_GOAL_NONE;
+
+	plat_goal_state_t g_goal_state = PLAT_GOAL_ACHIEVED;
+
+	struct pv_group *g, *tmp;
+	dl_list_for_each_safe(g, tmp, &s->groups, struct pv_group, list)
+	{
+		if (group == g)
+			break;
+
+		g_goal_state = pv_group_check_goals(g);
```

Comment:
Here you don't need to do something like this?
```
if (g_goal_state > s_goal_state)
			s_goal_state = g_goal_state;
```

Reply:
that was actually how it was before. In the case of pv_state_check_goals that's OK, as any timeout in the state revision means we have to rollback but, in the case of pv_state_check_goal_prev_group, its not OK, as any timeout in the sate revision will override the return value of the function and ignore the value that we want, which is just the one from the previous group of the group we are evaluating